data_safe_haven/validators/validators.py

```diff
@@ -18,16 +18,75 @@ def aad_guid(aad_guid: str) -> str:
 
 
 def azure_location(azure_location: str) -> str:
-    if not re.match(r"^[a-z]+[0-9]?[a-z]*$", azure_location):
+    # Generate a list of locations with the following command:
+    # `az account list-locations --query "[?metadata.regionType == 'Physical'].name"`
+    locations = [
+        "australiacentral",
+        "australiacentral2",
+        "australiaeast",
+        "australiasoutheast",
+        "brazilsouth",
+        "brazilsoutheast",
+        "brazilus",
+        "canadacentral",
+        "canadaeast",
+        "centralindia",
+        "centralus",
+        "centraluseuap",
+        "eastasia",
+        "eastus",
+        "eastus2",
+        "eastus2euap",
+        "eastusstg",
+        "francecentral",
+        "francesouth",
+        "germanynorth",
+        "germanywestcentral",
+        "israelcentral",
+        "italynorth",
+        "japaneast",
+        "japanwest",
+        "jioindiacentral",
+        "jioindiawest",
+        "koreacentral",
+        "koreasouth",
+        "mexicocentral",
+        "northcentralus",
+        "northeurope",
+        "norwayeast",
+        "norwaywest",
+        "polandcentral",
+        "qatarcentral",
+        "southafricanorth",
+        "southafricawest",
+        "southcentralus",
+        "southeastasia",
+        "southindia",
+        "spaincentral",
+        "swedencentral",
+        "switzerlandnorth",
+        "switzerlandwest",
+        "uaecentral",
+        "uaenorth",
+        "uksouth",
+        "ukwest",
+        "westcentralus",
+        "westeurope",
+        "westindia",
+        "westus",
+        "westus2",
+        "westus3",
+    ]
+    if azure_location not in locations:
         msg = "Expected valid Azure location, for example 'uksouth'."
         raise ValueError(msg)
     return azure_location
 
 
 def azure_subscription_name(subscription_name: str) -> str:
     # https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules
-    if not re.match(r"^[a-zA-Z][a-zA-Z0-9\- ]+[a-zA-Z0-9]$", subscription_name):
-        msg = "Azure subscription names can only contain alphanumeric characters and hyphens.\n They must start with a letter and end with an alphanumeric character."
+    if not re.match(r"^[a-zA-Z0-9\- \[\]]+$", subscription_name):
+        msg = "Azure subscription names can only contain alphanumeric characters, spaces and particular special characters."
```

Comment:
I think this now accepts invalid subscription names.

[docs](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules)

Reply:
Let's go with the more permissive rules then.

In fact, how worried are we about invalid subscription names? We don't create subscriptions do we? If someone enters an invalid name the worst that can happen is it isn't found.

Reply:
Good point - we should bear in mind that we might not want to validate this at all.